pkg/service/async.go

```diff
@@ -137,46 +138,48 @@ func processAsyncFilterAndAdd(ctx context.Context, wg *sync.WaitGroup) {
 					}
 					device.Origin = millis
 					device.Description = d.Description
-					_, err := common.DeviceClient.Add(ctx, device)
+
+					dc := s.edgexClients.DeviceClient
```

Comment:
Is there a reason you're creating an new instance of `dc` each time through the loop? Why not use `s.edgeClients.DeviceClient `directly?

Reply:
updated